tests/mock_slave.hpp

```diff
@@ -35,11 +35,11 @@ class mock_slave : public cosim::slave
 
     // Constructor that takes time-dependent functions
     explicit mock_slave(
-        std::function<double(cosim::time_point, double)> realOp = nullptr,
-        std::function<int(cosim::time_point, int)> intOp = nullptr,
-        std::function<bool(cosim::time_point, bool)> boolOp = nullptr,
-        std::function<std::string(cosim::time_point, std::string_view)> stringOp = nullptr,
-        std::function<void(cosim::time_point)> stepAction = nullptr)
+        std::function<double(cosim::time_point, cosim::duration, double)> realOp = nullptr,
+        std::function<int(cosim::time_point, cosim::duration, int)> intOp = nullptr,
+        std::function<bool(cosim::time_point, cosim::duration, bool)> boolOp = nullptr,
+        std::function<std::string(cosim::time_point, cosim::duration, std::string_view)> stringOp = nullptr,
+        std::function<void(cosim::time_point, cosim::duration)> stepAction = nullptr)
```

Comment:
I remember we added this `cosim::duration` previously for ecco. But I think then these functions do not really emulate `do_step` in FMI spec, would this be conceptually okay?

One way to avoid this is to store the previous time_points in the test script (that uses mock_slave) and pass them as refs and compute duration within lambda.

Reply:
I guess having the option to export & import the current step size would be required eventually, if we ever want to combine ECCO with state saving.